programs/drift/src/instructions/lp_pool.rs

```diff
@@ -100,6 +105,103 @@ pub fn handle_update_constituent_target_weights<'c: 'info, 'info>(
     Ok(())
 }
 
+pub fn handle_update_lp_pool_aum<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, UpdateLPPoolAum<'info>>,
+) -> Result<()> {
+    let mut lp_pool = ctx.accounts.lp_pool.load_mut()?;
+    let state = &ctx.accounts.state;
+
+    let slot = Clock::get()?.slot;
+
+    let remaining_accounts = &mut ctx.remaining_accounts.iter().peekable();
+
+    let AccountMaps {
+        perp_market_map: _,
+        spot_market_map,
+        mut oracle_map,
+    } = load_maps(
+        remaining_accounts,
+        &MarketSet::new(),
+        &MarketSet::new(),
+        slot,
+        Some(state.oracle_guard_rails),
+    )?;
+
+    let constituent_map = ConstituentMap::load(&ConstituentSet::new(), remaining_accounts)?;
+
+    msg!("{}", constituent_map.0.len());
+    msg!("{}", lp_pool.constituents);
+
+    validate!(
+        constituent_map.0.len() == lp_pool.constituents as usize,
+        ErrorCode::WrongNumberOfConstituents,
+        "Constituent map length does not match lp pool constituent count"
+    )?;
+
+    let mut aum: u128 = 0;
+    for i in 0..lp_pool.constituents as usize {
+        let mut constituent = constituent_map.get_ref_mut(&(i as u16))?;
+        let spot_market = spot_market_map.get_ref(&constituent.spot_market_index)?;
+
+        let oracle_data = oracle_map.get_price_data_and_validity(
+            MarketType::Spot,
+            constituent.spot_market_index,
+            &spot_market.oracle_id(),
+            spot_market.historical_oracle_data.last_oracle_price_twap,
+            spot_market.get_max_confidence_interval_multiplier()?,
+        )?;
+
+        let oracle_price = {
+            if !is_oracle_valid_for_action(oracle_data.1, Some(DriftAction::UpdateLpPoolAum))? {
+                msg!(
+                    "Oracle data for spot market {} is invalid. Skipping update",
+                    spot_market.market_index,
+                );
+                if slot - constituent.last_oracle_slot > 400 {
+                    i64::MAX
+                } else {
+                    constituent.last_oracle_price
+                }
+            } else {
+                oracle_data.0.price
+            }
+        };
+
+        if oracle_price == i64::MAX {
+            return Err(ErrorCode::OracleTooStaleForLPAUMUpdate.into());
+        }
+
+        constituent.last_oracle_price = oracle_price;
+        constituent.last_oracle_slot = slot;
+
+        let token_amount = constituent.spot_balance.get_token_amount(&spot_market)?;
```

Comment:
believe we'll want to load token account associated with the constituent and then add the borrow-lend position balance

Reply:
yeah he's already using `constituent.get_full_balance` below